example/trigger/main.tf

```diff
@@ -3,37 +3,36 @@ provider "honeycombio" {
   api_key = "<your API key>"
 }
 
+data "honeycombio_query" "query" {
+  calculation {
+    op     = "AVG"
+    column = "duration_ms"
+  }
+
+  filter {
+    column = "trace.parent_id"
+    op     = "does-not-exist"
+  }
+
+  filter {
+    column = "app.tenant"
+    op     = "="
+    value  = "ThatSpecialTenant"
+  }
+
+  filter_combination = "AND"
+
+  # also supported: breakdowns
+}
+
 resource "honeycombio_trigger" "trigger" {
   name        = "Requests are slower than usuals"
   description = "Average duration of all requests for ThatSpecialTenant for the last 15 minutes."
   dataset     = "<your dataset>"
 
   disabled = false
 
-  query {
-    # exactly one calculation is required
-    calculation {
-      op     = "AVG"
-      column = "duration_ms"
-    }
-
-    # zero or more filter blocks
-    filter {
-      column = "trace.parent_id"
-      op     = "does-not-exist"
-    }
-
-    filter {
-      column = "app.tenant"
-      op     = "="
-      value  = "ThatSpecialTenant"
-    }
-
-    # this can be ommited, AND is the default
-    filter_combination = "AND"
-
-    # also supported: breakdowns
-  }
+  query = data.honeycombio_query.query.rendered
```

Comment:
So if we do ever decide that we want to bring the embedded DSL back as an option, it might make sense to call this `query_json`... could also be a useful hint on what the structure should look like.

Reply:
maybe not, I could go either way on that one.
It definitely feels simpler if JSON is the only way to do it and lines up with the way `data aws_iam_policy_document` works, which is part of probably the most widely used provider out therte

Reply:
Good call, I'll change the field on trigger to `query_json` and the output of query to `json`. Then you get:
```hcl
data "honeycombio_query" "my_query" {
  # ...
}

resource "honeycombio_trigger" "trigger" {
  # ...

  query_json = data.honeycombio_query.my_query.json
}
```
This should make it clear how to connect the pieces.

I propose only supporting `query_json` (at least for now). Using the `honeycombio_query` should be fairly straight-forward and it's great for sharing queries between triggers and boards.
I'm wary about re-adding the nested DSL since that was fairly complicated and a lot of code to maintain.